network/stream/stream.go

```diff
@@ -318,6 +318,14 @@ func (r *Registry) clientCreateSendWant(ctx context.Context, p *Peer, stream ID,
 	}
 
 	p.mtx.Lock()
+	s := fmt.Sprintf("%s_%t", stream.String(), head)
+	if v, ok := p.openGetRange[s]; ok {
+		p.logger.Warn("batch already requested, skipping", "stream", stream, "head", head, "from", from, "to", to, "existing ruid", v)
+		return nil
```

Comment:
```suggestion
              p.mtx.Unlock()
		return nil
```

Reply:
It also means that this is not tested as all tests are passing with clear deadlock possibility on the improvement that is implemented in this pr. It would be good to have tests for this improvement at some point.